docs/src/extend/selectors.md

```diff
@@ -46,7 +46,9 @@ The following selectors are supported:
 * following sibling: `VariableDeclaration ~ VariableDeclaration`
 * adjacent sibling: `ArrayExpression > Literal + SpreadElement`
 * negation: `:not(ForStatement)`
+* has: `:has(ForStatement)`
 * matches-any: `:matches([attr] > :first-child, :last-child)`
+* subject indicator: `!ForStatement > [attr="foo"]`
```

Comment:
As for this one, I couldn't find any discussions on why it wasn't documented in ESLint, but it could be because it doesn't work quite well: https://github.com/estools/esquery/issues/60